accounts-db/benches/bench_lock_accounts.rs

```diff
@@ -0,0 +1,98 @@
+#![feature(test)]
+#![allow(clippy::arithmetic_side_effects)]
+
+extern crate test;
+
+use {
+    criterion::{criterion_group, criterion_main, Criterion},
+    rand::{seq::SliceRandom, thread_rng},
+    solana_accounts_db::{accounts::Accounts, accounts_db::AccountsDb},
+    solana_sdk::{
+        account::AccountSharedData,
+        instruction::{AccountMeta, Instruction},
+        pubkey::Pubkey,
+        system_program,
+        transaction::{SanitizedTransaction, Transaction, MAX_TX_ACCOUNT_LOCKS},
+    },
+    std::sync::Arc,
+};
+
+// simultaneous transactions locked
+const BATCH_SIZES: [usize; 3] = [1, 32, 64];
+
+// locks acquired per transaction
+const LOCK_COUNTS: [usize; 2] = [2, 64];
+
+// largest batch size * largest lock count * 2
+const ACCOUNTS_DB_SIZE: usize = 8192;
+
+fn create_test_data(batch_size: usize, lock_count: usize) -> (Accounts, Vec<SanitizedTransaction>) {
+    let accounts_db = AccountsDb::new_single_for_tests();
+    let accounts = Accounts::new(Arc::new(accounts_db));
+
+    let mut all_account_pubkeys = vec![];
+    let mut transactions = vec![];
+
+    for _ in 0..batch_size {
+        let mut account_metas = vec![];
+
+        // `lock_accounts()` distinguishes writable from readonly, so give transactions an even split
+        // signer doesnt matter for locking but `sanitize()` expects to see at least one
+        for i in 0..lock_count {
+            let pubkey = Pubkey::new_unique();
+
+            let account_meta = if i % 2 == 0 {
+                AccountMeta::new(pubkey, true)
+            } else {
+                AccountMeta::new_readonly(pubkey, false)
+            };
+
+            account_metas.push(account_meta);
+            all_account_pubkeys.push(pubkey);
+        }
+
+        let instruction = Instruction::new_with_bincode(system_program::id(), &(), account_metas);
+        let transaction = Transaction::new_with_payer(&[instruction], None);
+
+        transactions.push(SanitizedTransaction::from_transaction_for_tests(
+            transaction,
+        ));
+    }
+
+    // fill out the rest so all batches have the same size accounts-db
+    for _ in all_account_pubkeys.len()..ACCOUNTS_DB_SIZE {
+        all_account_pubkeys.push(Pubkey::new_unique());
+    }
+    assert_eq!(all_account_pubkeys.len(), ACCOUNTS_DB_SIZE);
+
+    // and shuffle before constructing accounts-db so lookup is random
+    all_account_pubkeys.shuffle(&mut thread_rng());
```

Comment:
shouldn't be looking up so I think no need for randomization here (or even the collection of all account keys?)